java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java

```diff
@@ -468,6 +615,39 @@ CompletableFuture<Double> zaddIncr(
      */
     CompletableFuture<String[]> zrange(String key, RangeQuery rangeQuery, boolean reverse);
 
+    /**
+     * Returns the specified range of elements in the sorted set stored at <code>key</code>.<br>
+     * <code>ZRANGE</code> can perform different types of range queries: by index (rank), by the
+     * score, or by lexicographical order.<br>
+     * To get the elements with their scores, see {@link #zrangeWithScores}.
+     *
+     * @see <a href="https://redis.io/commands/zrange/">redis.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param rangeQuery The range query object representing the type of range query to perform.<br>
+     *     <ul>
+     *       <li>For range queries by index (rank), use {@link RangeByIndex}.
+     *       <li>For range queries by lexicographical order, use {@link RangeByLex}.
+     *       <li>For range queries by score, use {@link RangeByScore}.
+     *     </ul>
+     *
+     * @param reverse If true, reverses the sorted set, with index 0 as the element with the highest
+     *     score.
+     * @return An array of elements within the specified range. If <code>key</code> does not exist, it
+     *     is treated as an empty sorted set, and the command returns an empty array.
+     * @example
+     *     <pre>{@code
+     * RangeByScoreBinary query1 = new RangeByScoreBinary(new ScoreBoundaryBinary(10), new ScoreBoundaryBinary(20));
+     * GlideString[] payload1 = client.zrange(gs("mySortedSet"), query1, true).get(); // Returns members with scores between 10 and 20.
+     * assert Arrays.equals(payload1, new GlideString[] {gs('member3'), gs('member2'), gs('member1')}); // Returns all members in descending order.
+     *
+     * RangeByScoreBinary query2 = new RangeByScoreBinary(InfScoreBoundBinary.NEGATIVE_INFINITY, new ScoreBoundaryBinary(3));
+     * GlideString[] payload2 = client.zrange(gs("mySortedSet"), query2, false).get();
+     * assert Arrays.equals(payload2, new GlideString[] {gs('member2'), gs('member3')}); // Returns members with scores within the range of negative infinity to 3, in ascending order.
```

Comment:
`assert Arrays.equals(payload2, new GlideString[] {gs("member2"), gs("member3")}); // Returns members with scores within the range of negative infinity to 3, in ascending order.`